routes.js

```diff
@@ -0,0 +1,327 @@
+import express from 'express';
+import path from 'path';
+import Joi from 'joi';
+import { requireJsonContent } from './middleware/index';
+import netflixData from './data/netflix-titles.json';
+
+const router = express.Router();
+const ITEMS_PER_PAGE = 20;
+let remainingPages = null;
+let totalPages = null;
+let movies = netflixData;
+
+// GET routes
+router.get('/', (req, res) => {
+  res.sendFile(path.join(__dirname + '/index.html'));
+});
+
+router.get('/movies', (req, res) => {
+  const queryPage = +req.query.page;
+  const queryYear = +req.query.year;
+  const queryDuration = +req.query.duration;
+  const queryActor = req.query.actor;
+
+  // Local copy of movies
+  let filteredMovies = movies;
+
+  // Query - Year
+  if (queryYear) {
+    filteredMovies = filteredMovies.filter(
+      movie => movie.release_year === queryYear
+    );
+  }
+
+  // Query - Actor
+  if (queryActor) {
+    filteredMovies = filteredMovies.filter(movie => {
+      let actors = movie.cast.split(', ');
+
+      // Check if current movie contains value in queryActor
+      const result = actors.filter(item => {
+        // Create regular expression with dynamic parameter
+        let regexp = new RegExp(
+          '\\b(' + queryActor.toLowerCase() + ')\\b',
+          'gi'
+        );
+
+        // Check for match
+        const match = item.toLowerCase().match(regexp);
+
+        // If actor is present in current movie item is returned
+        if (match) {
+          return item;
+        } else {
+          return;
+        }
+      });
+
+      // Return current movie if actor is found
+      if (result.length > 0) {
+        return movie;
+      }
+    });
+  }
+
+  // Query - Duration
+  if (queryDuration) {
+    filteredMovies = filteredMovies.filter(movie => {
+      const duration = movie.duration.split(' ');
+      if (
+        +duration[0] === queryDuration &&
+        duration[1].toLowerCase() !== 'seasons'
+      ) {
+        return movie;
+      }
+    });
+  }
+
+  // Query - Page
+  if (queryPage) {
+    const startIndex = queryPage * ITEMS_PER_PAGE - ITEMS_PER_PAGE; // Page 1 starts at index 0
+    const endIndex = startIndex + ITEMS_PER_PAGE;
+    totalPages = Math.ceil(filteredMovies.length / ITEMS_PER_PAGE);
+    remainingPages = totalPages - queryPage;
+    filteredMovies = filteredMovies.slice(startIndex, endIndex);
+  }
```

Comment:
I think this should be the default behavior - not returning all items. If this were a real API using a database, it would be very heavy to return all items all the time, so it's good to limit by default.